### PR TITLE
OVERHAUL PT. 5: SMGS REMADE

### PR DESCRIPTION
**Pull request by mazzinsanity**
### THIS PR IS PART OF THE ARMORY OVERHAUL PROJECT. SPRITES BY ELECTRICPANTS (@slatrhaus) CODE BY MAZZINSANITY (@mazzinsanity)
- Full SMG resprite
- Entirely new SMG stats (14mm unchanged)
- Rewritten SMG names and descriptions
- A new superhigh tier weapon, the 9mm machine gun - please welcome the Calico to the weapon roster
- Loadouts using SMGs have been rebalanced to fit the new SMG stats
- A new Explorer loadout
- New 9mm drum mags - obtainable as valuable bunker loot
- Light and Medium pistol calibers now have damage dropoff
- P90 rechambered in 5mm
- 14mm SMG in the Fox Vexillarius loadout replaced by the new P90
- Both the 10mm Carl Gustaf and its .45 ACP variant have been completely removed
- Both the craftable Micro-Uzi and Thompson Chicago Typewriter have been completely removed
- Tribal White-Legs loadout-specific Storm Drum Thompson variant has been completely removed

Pictures of the new SMG sprites attached:
image

Note the different sprites when a drum magazine is loaded.
image

New SMG stat details:

Selecting semi-auto mode now reduces spread and recoil by ~20% and increases the fire delay. Additionally, on SMGs that fit in bags (Normal size,) it allows for dual-wielding and one-handed shooting.

silenced .22 SMG (Formerly American 180)
- .22 LR, Supressed, Bullethose, 180 Round drum, Bulky, Low-mid tier
- fire_delay = 2
- autofire_shot_delay = 1
- slowdown = 0.25
- spread = 25

9mm submachine gun (Formerly Uzi)
- 9mm, Bullethose, Dual-wieldable on semi-auto, 32/50 Rounds, Normal, High-mid tier
- fire_delay = 2
- autofire_shot_delay = 1
- slowdown = 0.12
- spread = 20
- extra_damage = -2	//14 DAM

advanced 9mm SMG (Formerly MP5)
- 9mm, Suppressed, Dual-wieldable on semi-auto, AP, Accurate, 32/50 Rounds, Normal, High-mid tier
- fire_delay = 3
- autofire_shot_delay = 2
- slowdown = 0.15
- spread = 6
- extra_damage = -2 (18 DAM)
- extra_penetration = 0.15 (15% AP)

9mm machine gun (Calico)
- 9mm, Bullethose, 70 rounds, Bulky, Superhigh tier
- fire_delay = 2
- autofire_shot_delay = 1
- slowdown = 0.3
- spread = 16

10mm submachine gun
- 10mm, Dual-wieldable on semi-auto, 30 Rounds, Normal, Mid tier
- fire_delay = 3
- autofire_shot_delay = 2
- slowdown = 0.15
- spread = 14

cheap .45 Auto SMG (Formerly Grease Gun)
- .45 ACP, Dual-wieldable on semi-auto, Normal, 30 Rounds
- fire_delay = 3
- autofire_shot_delay = 2
- slowdown = 0.15
- spread = 18
- extra_damage = -4 (20 DAM)

.45 Auto submachine gun (Formerly Tommy Gun)
- .45 ACP, 30/50 Rounds, Bulky, High-mid tier
- fire_delay = 3
- autofire_shot_delay = 2
- slowdown = 0.3
- spread = 13

5mm submachine gun (Formerly P90)
- 5mm, AP, 50 Rounds, Bulky, Superhigh tier
- fire_delay = 3
- autofire_shot_delay = 2 
- slowdown = 0.2 
- spread = 13
- extra_damage = -5 (15 DAM)
- extra_penetration = -0.15 (30% AP)

14mm submachine gun
- 14mm, AP, 21/27 rounds, Bulky, Superhigh tier
- fire_delay = 4
- autofire_shot_delay = 3
- slowdown = 0.4
- spread = 12
- extra_penetration = 0.05 (5% AP)

Pistol caliber damage falloff details:

- .38, .22 LR, 9mm - Lose 5% damage per tile travelled (50% damage will be lost at 10 tiles traveled)
- 10mm, .45 ACP - Lose 2.5% damage per tile travelled (25% damage will be lost at 10 tiles traveled)
- Other pistol calibers unchanged

Loadout details:
- "Grunt Enforcer" Khan loadout - Grease gun replaced by 10mm SMG
- "Warlord Centurion" Centurion loadout - 10mm SMG replaced by Uzi (drum mag included)
- "Aspiring Prime Decanus" Prime Decanus loadout - .357 revolver replaced by MP5
- "Frontier Decanus" Recruit Decanus loadout - Uzi replaced by 10mm SMG
- "Desert Fox" Vexillarius loadout - 14mm SMG replaced by P90
- New "Stalker" Explorer loadout - MP5 and a smoke grenade
- "Flanker" Veteran Legionary loadout - 10mm SMG replaced by Uzi
- "CQB Patrol Ranger" NCR Ranger loadout - 10mm SMG replaced by MP5
- "Combat Sapper" and "Construction Specialist" NCR Combat Engineer loadout - Uzi replaced by 10mm SMG
- "Combat Medic" NCR Combat Medic loadout - Uzi replaced by 10mm SMG
- "White Legs Storm-Drummer" Tribal loadout - Storm Drum (Thomspon variant) replaced by 10mm SMG

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -226,37 +226,36 @@
 
 //Uzi			Keywords: 9mm, Dual-wieldable, Bullethose, 32/50 rounds
 /obj/item/gun/ballistic/automatic/smg/uzi
-	name = "Uzi"
-	desc = "A lightweight, burst-fire submachine gun, for when you really want someone dead. Uses 9mm rounds."
+	name = "9mm submachine gun"
```
Comment (hubertjones) [inline code/modules/projectiles/guns/ballistic/automatic.dm:229]: uzi nine millimeeta

**Comment by CloverIsLucky** [comment]
The P90 is actually a reskinned chinese LMG for the vex 💀💀💀💀

**Comment by electrical-pants** [comment]
These sprites were ALL mine! ALL of them!

chicanery

I am not crazy! I know I made those sprites! I knew it was mine. One after another. As if I could ever make such a mistake. Never. Never! I just – I just couldn't prove it. He – he covered his tracks, he got that idiot on the github to lie for him. You think this is something? You think this is bad? This? This chicanery? He's done worse. The Legion medicine rework! Are you telling me that bitters just happen to be overpowered? No! He orchestrated it! Mazz! He broke it! And I saved him! And I shouldn't have. I took him into my own server! What was I thinking? He'll never change. He'll never change! Ever since he was 9, always the same! Couldn't keep his hands out of the cash drawer! But not our Mazz! Couldn't be precious Mazz! Stealing them blind! And he gets to be a coder!? What a sick joke! I should've stopped him when I had the chance! And you – you have to stop him! You-

**Comment by Troontjelolo** [comment]
> These sprites were ALL mine! ALL of them!
> 
> chicanery [ chicanery ](https://private-user-images.githubusercontent.com/98723149/291030869-f5ceac64-b1a9-44f3-a1f1-afa8b323995b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5ODUwNjUsIm5iZiI6MTcwMjk4NDc2NSwicGF0aCI6Ii85ODcyMzE0OS8yOTEwMzA4NjktZjVjZWFjNjQtYjFhOS00NGYzLWExZjEtYWZhOGIzMjM5OTViLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDExMTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MGE2ZTQ4YTlkNWZjNWQ0NjIzNjAwYWNmOGE3ZWVlNTc2YzMzNmI0ZjIwOWEyMjAyMzZmYzJhZjc1ZjlhMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.T-2QidDXX55xDRZw_hJQ1dcC0Z-XsV3iz8HGbYa2790) [ ](https://private-user-images.githubusercontent.com/98723149/291030869-f5ceac64-b1a9-44f3-a1f1-afa8b323995b.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5ODUwNjUsIm5iZiI6MTcwMjk4NDc2NSwicGF0aCI6Ii85ODcyMzE0OS8yOTEwMzA4NjktZjVjZWFjNjQtYjFhOS00NGYzLWExZjEtYWZhOGIzMjM5OTViLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDExMTkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MGE2ZTQ4YTlkNWZjNWQ0NjIzNjAwYWNmOGE3ZWVlNTc2YzMzNmI0ZjIwOWEyMjAyMzZmYzJhZjc1ZjlhMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.T-2QidDXX55xDRZw_hJQ1dcC0Z-XsV3iz8HGbYa2790)
> 
> I am not crazy! I know I made those sprites! I knew it was mine. One after another. As if I could ever make such a mistake. Never. Never! I just – I just couldn't prove it. He – he covered his tracks, he got that idiot on the github to lie for him. You think this is something? You think this is bad? This? This chicanery? He's done worse. The Legion medicine rework! Are you telling me that bitters just happen to be overpowered? No! He orchestrated it! Mazz! He broke it! And I saved him! And I shouldn't have. I took him into my own server! What was I thinking? He'll never change. He'll never change! Ever since he was 9, always the same! Couldn't keep his hands out of the cash drawer! But not our Mazz! Couldn't be precious Mazz! Stealing them blind! And he gets to be a coder!? What a sick joke! I should've stopped him when I had the chance! And you – you have to stop him! You-

image

**Review by Stutternov — APPROVED** [review]

**Review by Stutternov — APPROVED** [review]

**Pull request merged**